packaging/amazon-linux-ami/ecs-init.spec

```diff
@@ -33,24 +34,93 @@ Requires:       iptables
+Provides:       bundled(golang(github.com/docker/docker/api/types/strslice))
+Provides:       bundled(golang(github.com/docker/docker/api/types/swarm))
+Provides:       bundled(golang(github.com/docker/docker/api/types/versions))
+Provides:       bundled(golang(github.com/docker/docker/opts))
 Provides:       bundled(golang(github.com/docker/docker/pkg/archive))
 Provides:       bundled(golang(github.com/docker/docker/pkg/fileutils))
+Provides:       bundled(golang(github.com/docker/docker/pkg/homedir))
+Provides:       bundled(golang(github.com/docker/docker/pkg/idtools))
 Provides:       bundled(golang(github.com/docker/docker/pkg/ioutils))
+Provides:       bundled(golang(github.com/docker/docker/pkg/jsonlog))
+Provides:       bundled(golang(github.com/docker/docker/pkg/jsonmessage))
+Provides:       bundled(golang(github.com/docker/docker/pkg/longpath))
+Provides:       bundled(golang(github.com/docker/docker/pkg/mount))
 Provides:       bundled(golang(github.com/docker/docker/pkg/pools))
 Provides:       bundled(golang(github.com/docker/docker/pkg/promise))
+Provides:       bundled(golang(github.com/docker/docker/pkg/stdcopy))
 Provides:       bundled(golang(github.com/docker/docker/pkg/system))
-Provides:       bundled(golang(github.com/docker/docker/pkg/units))
-Provides:       bundled(golang(github.com/docker/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar))
+Provides:       bundled(golang(github.com/docker/docker/pkg/term))
+Provides:       bundled(golang(github.com/docker/docker/pkg/term/windows))
+Provides:       bundled(golang(github.com/docker/go-connections/nat))
+Provides:       bundled(golang(github.com/docker/go-units))
 Provides:       bundled(golang(github.com/fsouza/go-dockerclient))
+Provides:       bundled(golang(github.com/go-ini/ini))
 Provides:       bundled(golang(github.com/golang/mock/gomock))
-Provides:       bundled(golang-logrus)
+Provides:       bundled(golang(github.com/jmespath/go-jmespath))
+Provides:       bundled(golang(github.com/Microsoft/go-winio))
+Provides:       bundled(golang(github.com/Nvveen/Gotty))
+Provides:       bundled(golang(github.com/opencontainers/go-digest))
+Provides:       bundled(golang(github.com/opencontainers/image-spec/specs-go))
+Provides:       bundled(golang(github.com/opencontainers/image-spec/specs-go/v1))
+Provides:       bundled(golang(github.com/opencontainers/runc/libcontainer/system))
+Provides:       bundled(golang(github.com/opencontainers/runc/libcontainer/user))
+Provides:       bundled(golang(github.com/pkg/errors))
 Provides:       bundled(golang(github.com/Sirupsen/logrus))
-Provides:       bundled(golang(github.com/cihub/seelog))
+Provides:       bundled(golang(golang.org/x/net/context))
+Provides:       bundled(golang(golang.org/x/net/context/ctxhttp))
+Provides:       bundled(golang(golang.org/x/sys/unix))
+Provides:       bundled(golang(golang.org/x/sys/windows))
 
+%global bundled_agent_version %{version}
```

Comment:
I don't see where `%{version}` is defined.  Should we read this from `ecs-init/VERSION`?

It might also be worthwhile passing `%{bundled_agent_version}` into the the build script (maybe as an argument or environment variable in the `%build` section) so we don't need to [maintain the constant](https://github.com/aws/amazon-ecs-init/blob/77120e36977944f8f78a3861b623aa53c3f11bcb/ecs-init/config/common.go#L36-L37) separately in the source anymore

Reply:
Version is [defined by the spec file](https://rpm-packaging-guide.github.io/#rpm-macros), we're tying this to the `ecs-init` Version until it diverges.

We could pass on the parameter into the build scripts, however externalizing that option would help the package configure `ecs-init`, either way it's something we can certainly consider in another iteration.

Reply:
I'm inclined to stick with this behavior rather than changing the way we invoke the build.